react/provider/ConfigProvider.tsx

```diff
@@ -18,71 +21,86 @@ const Config = React.createContext<MappSettingsProvider>({
   isSaving: false,
   setConfig: () => {},
   updateConfig: () => {},
-});
+})
 
-export const ConfigProvider: React.FC = (props) => {
-  const { data: dataSettingsSchema } = useQuery(settingsSchema);
-  const [saveSettingsMutation] = useMutation(saveMappSettings);
+export const ConfigProvider: React.FC = props => {
+  const { data: dataSettingsSchema } = useQuery(settingsSchema)
+  const [saveSettingsMutation] = useMutation(saveMappSettings)
   const [config, setConfig] = useState<MappSettings>({
-    tiId: "",
-    tiResponder: "responder.wt-safetag.com",
-    acId: "",
-    acM: "",
-  });
+    tiId: '',
+    tiResponder: 'responder.wt-safetag.com',
+    acId: '',
+    acM: '',
+    engageApiUrl: '',
+    engageIntegrationId: '',
+    engageSecret: '',
+  })
 
-  const [isSaving, setIsSaving] = useState(false);
-  const [configLoading, setConfigLoading] = useState(true);
+  const [isSaving, setIsSaving] = useState(false)
+  const [configLoading, setConfigLoading] = useState(true)
 
   const saveSettings = async () => {
-    setIsSaving(true);
-    const settings = {...config};
-    if(settings.acId === "") {
-      settings.acId = '0';
+    setIsSaving(true)
+    const settings = { ...config }
+
+    if (settings.acId === '') {
+      settings.acId = '0'
     }
-    if(settings.acM === "") {
-      settings.acM = '0';
+
+    if (settings.acM === '') {
+      settings.acM = '0'
     }
-    if(settings.tiId === "") {
-      settings.tiId = '0';
+
+    if (settings.tiId === '') {
+      settings.tiId = '0'
     }
-    if(settings.tiResponder === "") {
-      settings.tiResponder = '0';
+
+    if (settings.tiResponder === '') {
+      settings.tiResponder = '0'
     }
+
     await saveSettingsMutation({
       variables: { settings: JSON.stringify(settings) },
-    });
-    setIsSaving(false);
-  };
+    })
+    setIsSaving(false)
+  }
 
-  const updateConfig: MappSettingsProvider["updateConfig"] = (newValue) => {
+  const updateConfig: MappSettingsProvider['updateConfig'] = newValue => {
     setConfig((oldConfig: MappSettings) => {
       return {
         ...oldConfig,
         ...newValue,
-      };
-    });
-  };
+      }
+    })
+  }
 
   useEffect(() => {
-    let settings = dataSettingsSchema?.appSettings?.message;
-    if(settings) {
-      settings = JSON.parse(settings);
-      if(settings.acId === "0") {
-        settings.acId = '';
-      }
-      if(settings.acM === "0") {
-        settings.acM = '';
-      }
-      if(settings.tiId === "0") {
-        settings.tiId = '';
-      }
-      if(settings.tiResponder === "0") {
-        settings.tiResponder = '';
-      }
```

Comment:
here yur new config props need to be added too. This is all this workaround for the bug: there must be a value saved, but we don't want to use that default value so we make it disappear here